### ISSUE TITLE
DuckDB 0.8.0 Pandas DataFrame and SQLite queries on small string columns now return large_string

### ISSUE THREAD

**Issue by d33bs** [duckdb/duckdb#7606]
### What happens?

Thanks so much for the amazing work on this project - it's tremendously helpful!

DuckDB 0.8.0 queries performed in Python on Pandas DataFrames and SQLite tables (via sqlite_scanner) are now returning large_string PyArrow data types by default. DuckDB 0.7.1 did not exhibit this behavior, so I wanted to raise a bug just in case. Thank you in advance for your assistance with this!

### To Reproduce

To observe this behavior, please run the following in both duckdb==0.7.1 (which yields "string" arrow types for the string values) and duckdb==0.8.0 (which yields "large_string" arrow types). 

```python
import sqlite3

import duckdb
import pandas as pd

# create a temporary sqlite connection
filepath = ".example.sqlite"

# statements for creating database with simple structure
create_stmts = [
    "DROP TABLE IF EXISTS tbl_a;",
    """
    CREATE TABLE tbl_a (
    col_integer INTEGER NOT NULL
    ,col_text TEXT
    ,col_blob BLOB
    ,col_real REAL
    );
    """,
]

# some example values to insert into the database
insert_vals = [1, "sample", b"sample_blob", 0.5]


# create the database and insert some data into it
with sqlite3.connect(filepath) as connection:
    for stmt in create_stmts:
        connection.execute(stmt)

    connection.execute(
        (
            "INSERT INTO tbl_a (col_integer, col_text, col_blob, col_real)"
            "VALUES (?, ?, ?, ?);"
        ),
        insert_vals,
    )


df = pd.read_sql(sql="select * from tbl_a", con=f"sqlite:///{filepath}")

print(
    "pandas dataframe result:",
    duckdb.connect()
    .execute(
    f"""
    /* perform query on dataframe */
    SELECT * FROM df;
    """
    )
    .arrow()
    .schema,
    sep="\n",
)
print(
    "sqlite_scanner result:",
    duckdb.connect()
    .execute(
    f"""
    /* install and load sqlite plugin for duckdb */
    INSTALL sqlite_scanner;
    LOAD sqlite_scanner;
    
    /* perform query on tbl_a table */
    SELECT * FROM sqlite_scan('{filepath}', 'tbl_a');
    """
    )
    .arrow()
    .schema,
    sep="\n",
)
```

### OS:

MacOS

### DuckDB Version:

0.8.0

### DuckDB Client:

Python

### Full Name:

Dave Bunten

### Affiliation:

University of Colorado

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by paultiq** [comment]
(linking to related discussion) https://github.com/duckdb/duckdb/discussions/7591

**Comment by carlopi** [comment]
This might have been solved by https://github.com/duckdb/duckdb/pull/7784, that should have been included in duckdb 0.8.1.
@d33bs, could you give it another go?

**Comment by d33bs** [comment]
Thank you @carlopi! I found that after upgrading to duckdb 0.8.1 this issue no longer occurs.
